types/account.go

```diff
@@ -52,27 +52,29 @@ func (acc EthAccount) EthAddress() ethcmn.Address {
 // TODO: remove on SDK v0.40
 
 // Balance returns the balance of an account.
-func (acc EthAccount) Balance() sdk.Int {
-	return acc.GetCoins().AmountOf(AttoPhoton)
+func (acc EthAccount) Balance(denom string) sdk.Int {
+	return acc.GetCoins().AmountOf(denom)
 }
 
-// SetBalance sets an account's balance of aphotons
-func (acc *EthAccount) SetBalance(amt sdk.Int) {
+// SetBalance sets an account's balance of the given coin denomination.
+//
+// CONTRACT: assumes the denomination is valid.
+func (acc *EthAccount) SetBalance(denom string, amt sdk.Int) {
 	coins := acc.GetCoins()
-	diff := amt.Sub(coins.AmountOf(AttoPhoton))
+	diff := amt.Sub(coins.AmountOf(denom))
 	switch {
 	case diff.IsPositive():
 		// Increase coins to amount
-		coins = coins.Add(NewPhotonCoin(diff))
+		coins = coins.Add(sdk.NewCoin(denom, diff))
 	case diff.IsNegative():
 		// Decrease coins to amount
-		coins = coins.Sub(sdk.NewCoins(NewPhotonCoin(diff.Neg())))
+		coins = coins.Sub(sdk.NewCoins(sdk.NewCoin(denom, diff.Neg())))
```

Comment:
is the `sdk.NewCoins` necessary here? the add case doesn't have it

Reply:
yeah, idk the reason of the inconsistency on the SDK. The Add function has a spread parameter while the Sub doesn't

Reply:
weird 🤔 okay

Reply:
yeah lol